test/test_mcmc.py

```diff
@@ -201,3 +202,30 @@ def model(data):
 
     if 'JAX_ENABLE_x64' in os.environ:
         assert hmc_states['p'].dtype == np.float64
+
+
+@pytest.mark.parametrize('algo', ['HMC', 'NUTS'])
+def test_pmap(algo):
+    if xla_bridge.device_count() == 1:
+        pytest.skip('pmap test requires device_count greater than 1.')
+
+    true_mean, true_std = 1., 2.
+    warmup_steps, num_samples = 1000, 8000
+
+    def potential_fn(z):
+        return 0.5 * np.sum(((z - true_mean) / true_std) ** 2)
+
+    init_kernel, sample_kernel = hmc(potential_fn, algo=algo)
+    init_params = np.array([0., -1.])
+    rngs = random.split(random.PRNGKey(0), 2)
+
+    init_kernel_pmap = pmap(lambda init_param, rng: init_kernel(
+        init_param, trajectory_length=9, num_warmup=warmup_steps, progbar=False, rng=rng))
+    init_states = init_kernel_pmap(init_params, rngs)
+
+    fori_collect_pmap = pmap(lambda hmc_state: fori_collect(num_samples, sample_kernel, hmc_state,
```

Comment:
This is pretty cool! Based on your observation, how much slower is this as compared to a single chain?

Reply:
It is a bit faster than a single chain for this test (17s vs 19s) (I guess by the same reason as we jit lax.fori_loop). It is hard to tell how the change will affect bigger models because intra-op parallelism in CPU is not useful for this simple model. We need to do benchmark for bigger models while support multi-chain in mcmc (in later PRs though).

Reply:
> It is a bit faster than a single chain for this test (17s vs 19s)

That's strange - you mean using pmap to draw 16000 samples is faster than using the default single chain to draw 8000 samples?

Reply:
Yes. pmap just replicates jaxpr across device, so its effect is similar to jit(fori_loop) I guess. FYI, I don't know why `jit(fori_loop)` is faster than `fori_loop`.

Reply:
Ahh..I see. So that's the thing - maybe we need to understand why `jit(fori_loop)` is faster than `fori_loop`. This doesn't exactly seem very reasonable, maybe we should ask JAX devs if this is expected.

Reply:
Haha, incidently, I just ask on Slack. :D